sadl3/com.ge.research.sadl.parent/com.ge.research.sadl.feature/feature.xml

```diff
@@ -46,6 +46,26 @@ SADL also includes POJava licensed under Apache License V2.0.
       <discovery label="Semantic Application Design Language Updates and Wiki" url="https://github.com/crapo/sadlos2"/>
    </url>
 
+   <includes
+         id="org.eclipse.xtext.runtime"
+         version="0.0.0"/>
+
+   <includes
+         id="org.eclipse.xtext.ui"
+         version="0.0.0"/>
+
+   <includes
+         id="org.eclipse.xtext.xbase"
+         version="0.0.0"/>
+
+   <includes
+         id="org.eclipse.xtext.xbase.lib"
+         version="0.0.0"/>
+
+   <includes
+         id="org.eclipse.xtext.xtext.ui"
+         version="0.0.0"/>
+
```

Comment:
I've looked over the changeset quickly and I think it looks great, especially the removal of the xtext version ranges from the MANIFEST.MF files.  That will make it easier to build and/or import SADL into future Xtext/Eclipse platforms.  

I'm curious about this particular change.  What is the purpose of this change?  Is the purpose to bundle the Xtext bundles that SADL was built with so that SADL can be installed into an Eclipse platform which doesn't have any Xtext bundles at all?  Or is the purpose to bundle the Xtext bundles that SADL was built with so that SADL can be installed into an Eclipse platform which has Xtext bundles older than 2.14.0 and therefore SADL needs to bring its own Xtext bundles just in case?

Reply:
> I'm curious about this particular change. What is the purpose of this change?

Thanks for asking. Let assume you have an Eclipse 2020-03 for DSL developers:
![Screen Shot 2020-10-27 at 12 51 04](https://user-images.githubusercontent.com/1405703/97325226-0b033f80-1873-11eb-8d81-072b1dc8fa47.jpg)

It comes with Xtext 2.21:
![Screen Shot 2020-10-27 at 12 51 11](https://user-images.githubusercontent.com/1405703/97325296-19515b80-1873-11eb-95c7-bcf8042baeb9.jpg)

With this additional `include` (Xtext) in the manifest, when installing the SADL p2 into Eclipse 2020-3 DSL, it will automatically force updating the Xtext version shipped with Eclipse:
![Screen Shot 2020-10-27 at 12 52 17](https://user-images.githubusercontent.com/1405703/97325499-5d446080-1873-11eb-9aad-61acb213f040.jpg)

So once the SADL p2 installation is done, you will see the Xtext version update in the underlying platform:
![Screen Shot 2020-10-27 at 12 52 27](https://user-images.githubusercontent.com/1405703/97325589-7220f400-1873-11eb-9c53-701cec7ecb10.jpg)

If we do not force the Xtext version update when installing the SADL p2, we might run into incompatible APIs, causing a runtime error. Such as [this](https://github.com/SemanticApplicationDesignLanguage/sadl/issues/486#issuecomment-665053330).

-----
Note, if you install the SADL p2 into any Eclipse for Java developers, it will just make Xtext only dependent on SADL:
![Screen Shot 2020-10-27 at 12 59 23](https://user-images.githubusercontent.com/1405703/97325811-a7c5dd00-1873-11eb-9fc1-2c6a5063dc6f.jpg)